test/langtools/jdk/javadoc/doclet/testRestricted/TestRestricted.java

```diff
@@ -42,6 +42,7 @@ public class TestRestricted extends JavadocTester {
     public final ToolBox tb;
     public static void main(String... args) throws Exception {
         var tester = new TestRestricted();
+        tester.setAutomaticCheckLinks(false);
```

Comment:
@hns I had to disable this check because otherwise the test framework will attempt (and fail) to resolve `../java.base/java/lang/doc-files/RestrictedMethods.html`. Is there a better way to do this w/o disabling the link checks?

Reply:
I'm not Hannes, but to my mind, the fact that you had to disable that check to make the test pass hints that it's a real problem rather than a minor inconvenience.

Basing this URL off `{@docRoot}` seems an incorrect thing to do:

    doclet.Restricted.url={@docRoot}/java.base/java/lang/doc-files/RestrictedMethods.html

It will only be `@docRoot` in the JDK, but not in any other code base, and certainly not in that test.

I don't know if restricted methods is a JDK-only feature, but for javadoc's sake, we should treat it like it's not. So the link should go to the same place where `{@link Object}` goes.

Reply:
> It will only be `@docRoot` in the JDK, but not in any other code base, and certainly not in that test.

Note: only JDK is allowed to have restricted methods. `@Restricted` is a non-exported JDK annotation. So 3rd party libraries cannot define restricted methods. So, how to render the javadoc for restricted methods is "our own" issue.

Reply:
Sorry for missing the mention. I agree with @pavelrappo that this doesn't look quite right. It's true that it is not usually an issue as the feature will not be used outside of Java SE APIs. But it's conceivable that builds API docs for JDK or Java SE modules without including `java.base`. I think the right solution would be to create the link only if `java.base`/`java.lang` is included in the documentation.